File: Subarray Sum Equals K.js
const subarraySum = (nums, k) => {
	var map = {0:1};
    var count = 0;
    var sum = 0;
    
    for(var x of nums){
        sum += x;
        if(map[sum-k]) count+=map[sum-k];
        map[sum] = (map[sum] || 0) + 1;
    }
    return count;
};

File: Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pref_sum = collections.Counter()
        pref_sum[0] = 1
        running_sum = 0
        result = 0

        for num in nums:
            running_sum += num
            result += pref_sum[running_sum-k]
            pref_sum[running_sum] += 1
        
        return result
